docs/source/miscellaneous/doc/adr/0014-cassandra-targeted-sweep.md

```diff
@@ -0,0 +1,283 @@
+    of the reference is the same as above, but the write_index is a negative number with an absolute value equal to the number of
+    dedicated rows we will use (number of writes divided by 100_000, rounded up). Then, put the cells into dedicated rows, where
+    the *timestamp_partition* is the start timestamp of the transaction (**not the fine partition!**), the metadata encodes that the
+    row is a dedicated row and the row's ordinal number, and the rest is the same as above.
+
+#### Reading from the Sweep Queue
+
+Reading from the sweep queue is done in the reverse order from writing. For a given shard, strategy, **minimum exclusive
+timestmap** (targeted sweep reads this from the *sweepProgressPerShard* table as described later), and **maximum exclusive
+timestamp** (for targeted sweep, this is the sweep timestamp), we do the following:
+
+  1. We want to locate the fine timestamp partition for the first row of *sweepableCells* that has entries greater than the minimum
+     exclusive timestamp *minTs*. Starting with the coarse partition of *minTs + 1*, we use `getRowsColumnRange` to check if there
+     is a cell in SweepableTimestamps satisfying the above condition. If not, increase the coarse partition and repeat until
+     either a candidate is found, or the coarse partition grows larger than the coarse partition of *maxTs - 1*, where *maxTs* is
+     the maximum exclusive timestamp.
+  2. If the latter occurs, we are guaranteed that there are no entries in *sweepableCells* for the shard and strategy and any of the
+     timestamps in the specified range.
+  3. Otherwise, we now have a fine timestamp partition that is effectively a reference to a row of SweepableCells that is expected
+     to contain at least one cell (this may not be true in degenerate cases where a thread writing or cleaning the queue dies during
+     the process). We now read cells from that row, with start timestamp greater than the last swept timestamp, referring to
+     dedicated rows as necessary, until we either finish the row, exceed the *maxTs*, or we have read 100_000 entries, plus any
+     additional entries to ensure we have read all the entries with the same start timestamp as the latest one.
+
+#### Cleaning the Sweep Queue
+
+Once an entire non-dedicated row of *sweepableCells* or a row of *sweepableTimestamps* are not needed anymore, we remove them as
+follows.
+
+1. Given a shard, strategy, and fine timestamp partition delete all entries from *sweepableCells* for the corresponding row. Note
+   that in Cassandra, an entire row can be deleted at once using a single tombstone.
+    - First, we read through the non-dedicated row to find all the references to dedicated rows and delete them.
+    - Then, delete the the non-dedicated row.
+2. Then, if necessary, given a coarse timestamp partition, also delete the row defined by the shard, strategy, and the coarse
+   timestamp partition in *sweepableTimestamps*.
+
+### Targeted Sweep
```

Comment:
I'd put this section above the technical queue implementation details

Reply:
I think now it makes more sense, so removed it altogether